pkg/networkmanager/mtu.jsx

```diff
@@ -36,11 +38,28 @@ export const MtuDialog = ({ connection, dev, settings }) => {
     const [dialogError, setDialogError] = useState(undefined);
     const [mode, setMode] = useState(!settings.ethernet.mtu ? "auto" : "custom");
     const [mtu, setMtu] = useState(settings.ethernet.mtu ? settings.ethernet.mtu : '');
+    const [mtuValidation, setMtuValidation] = useState("");
 
-    const onSubmit = (ev) => {
-        const mtuNew = mode == 'auto' ? 0 : parseInt(mtu, 10);
+    const handleMtuChange = (_event, mtuChange) => {
+        const mtuNew = parseInt(mtuChange, 10);
         if (isNaN(mtuNew) || mtuNew < 0) {
-            setDialogError(_("MTU must be a positive number"));
+            setMtuValidation(_("MTU must be a positive number"));
+        } else {
+            setMtuValidation("");
+        }
+        if (mtuChange !== "" && mode === "auto") {
+            setMode("custom");
+        } else if (mtuChange === "" && mode === "custom") {
+            setMode("auto");
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22272-62b285ca-20251031-142546-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)